tests/components/zwave_js/test_config_flow.py

```diff
@@ -138,35 +137,36 @@ def mock_addon_setup_time():
 
 
 @pytest.fixture(name="serial_port")
-def serial_port_fixture() -> ListPortInfo:
+def serial_port_fixture() -> usb.USBDevice:
     """Return a mock serial port."""
-    port = ListPortInfo("/test", skip_link_detection=True)
-    port.serial_number = "1234"
-    port.manufacturer = "Virtual serial port"
-    port.device = "/test"
-    port.description = "Some serial port"
-    port.pid = 9876
-    port.vid = 5678
-
-    return port
+    return usb.USBDevice(
+        serial_number="1234",
+        manufacturer="Virtual serial port",
+        device="/test",
+        description="Some serial port",
+        pid=9876,
+        vid=5678,
+    )
 
 
 @pytest.fixture(name="mock_list_ports", autouse=True)
 def mock_list_ports_fixture(serial_port) -> Generator[MagicMock, None, None]:
     """Mock list ports."""
-    with patch(
-        "homeassistant.components.zwave_js.config_flow.list_ports.comports"
-    ) as mock_list_ports:
-        another_port = copy(serial_port)
-        another_port.device = "/new"
-        another_port.description = "New serial port"
-        another_port.serial_number = "5678"
-        another_port.pid = 8765
-        no_vid_port = copy(serial_port)
-        no_vid_port.device = "/no_vid"
-        no_vid_port.description = "Port without vid"
-        no_vid_port.serial_number = "9123"
-        no_vid_port.vid = None
+    with patch("homeassistant.components.usb.list_serial_ports") as mock_list_ports:
```

Comment:
We should patch the library and not our code.

Reply:
PySerial's `comports()` and `ListPortInfo` types are hidden from components by this PR so most will not be directly depending on the PySerial package at all. Instead, the USB component and its `USBDevice` type are used. In the end, the components only really care about prompting the user to pick a serial port.

It seems better to me to more thoroughly unit test `usb.list_serial_ports` within the USB integration and then just assume `homeassistant.components.usb.list_serial_ports` works when testing specific integrations.

Reply:
Where are those tests?

Reply:
I haven't written any yet, it's a draft PR 😄. I'm mainly looking for architectural feedback at this point.